File: Misc. Algos/Recursive_algos/towerofhanoi.c
// Tower of Hanoi
// code by Akshat Aryan
#include <stdio.h>

void tower(int n, char *src, char *aux, char *dest){
    if(n == 1){
        printf("Move Disk from %s to %s", src, dest);
    }
    else{
        tower(n - 1, src, dest, aux);
        printf("Move Disk from %s to %s", src, dest);
        tower(n - 1, aux, src, dest);
    }
}

int main(){
    int n;
    printf("Enter the number of disks:");
    scanf("%d", &n);
    printf("Follow these steps:\n");
    tower(n, "Source", "Auxiliary", "Destination");
    return 0;
}

File: Misc. Algos/Recursive_algos/towerofhanoi.py
# Tower of Hanoi
# code by Akshat Aryan

def tower(n, src, aux, dest):
    if n == 1:
        print("Move Disk from", src, "to", dest)
    else:
        tower(n - 1, src, dest, aux)
        print("Move Disk from", src, "to", dest)
        tower(n - 1, aux, src, dest)
        
if(__name__ == "__main__"):
    n = int(input("Enter the number of disks:"))
    print("Follow these steps:")
    tower(n, "Source", "Auxiliary", "Destination")